tests/test_benchmark.py

```diff
@@ -5,30 +5,152 @@
 from timemachine.md import builders, minimizer
 from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.testsystems.dhfr import setup_dhfr
+from timemachine.testsystems.relative import setup_hif2a_ligand_pair
 
 
-def recenter(conf, box):
+@pytest.fixture(autouse=True, scope="module")
+def hi2fa_test_frames():
+    return generate_hif2a_frames(100, 1, seed=2022)
 
-    new_coords = []
 
-    periodicBoxSize = box
+def generate_hif2a_frames(n_frames: int, frame_interval: int, seed=None, barostat_interval: int = 5, hmr: bool = True):
+    rfe = setup_hif2a_ligand_pair("smirnoff_1_1_0_ccc.py")
+    mol_a, mol_b = rfe.mol_a, rfe.mol_b
+
+    # build the protein system.
+    host_system, host_coords, _, _, host_box, _ = builders.build_protein_system("tests/data/hif2a_nowater_min.pdb")
+
+    unbound_potentials, sys_params, masses = rfe.prepare_host_edge(rfe.ff.get_ordered_params(), host_system)
+    min_host_coords = minimizer.minimize_host_4d([mol_a, mol_b], host_system, host_coords, rfe.ff, host_box)
+    x0 = rfe.prepare_combined_coords(min_host_coords)
+    ligand_idxs = np.arange(len(host_coords), len(x0), dtype=np.int32)
+
+    temperature = 300
+    pressure = 1.0
+    lamb = 0.0
+
+    harmonic_bond_potential = unbound_potentials[0]
+    bond_list = get_bond_list(harmonic_bond_potential)
+    if hmr:
+        dt = 2.5e-3
+        masses = apply_hmr(masses, bond_list)
+    if seed is None:
+        seed = np.random.randint(np.iinfo(np.int32).max)
+    intg = LangevinIntegrator(temperature, dt, 1.0, np.array(masses), seed).impl()
+
+    bps = []
+
+    for potential, params in zip(unbound_potentials, sys_params):
+        bps.append(potential.bind(params).bound_impl(precision=np.float32))  # get the bound implementation
+
+    baro_impl = None
+    if barostat_interval > 0:
+        group_idxs = get_group_indices(bond_list)
+        baro = MonteCarloBarostat(
+            x0.shape[0],
+            pressure,
+            temperature,
+            group_idxs,
+            barostat_interval,
+            seed,
+        )
+        baro_impl = baro.impl(bps)
+
+    ctxt = custom_ops.Context(
+        x0,
+        np.zeros_like(x0),
+        host_box,
+        intg,
+        bps,
+        barostat=baro_impl,
+    )
+    steps = n_frames * frame_interval
+    _, coords, boxes = ctxt.multiple_steps(np.ones(steps) * lamb, 0, frame_interval)
+    assert coords.shape[0] == n_frames, f"Got {coords.shape[0]} frames, expected {n_frames}"
+    return unbound_potentials, sys_params, coords, boxes, ligand_idxs
+
+
+def benchmark_potential_over_coords(
+    label,
+    potential,
+    precision,
+    params,
+    coords,
+    boxes,
+    lamb,
+    verbose=True,
+    num_batches=5,
+):
+    unbound = potential.unbound_impl(precision=precision)
+    start = time.time()
+    batch_times = []
+    frames_per_batch = coords.shape[0]
+    for _ in range(num_batches):
```

Comment:
Is it worth extracting the timing logic into a helper function, since it's replicated in each benchmark?

[timeit](https://docs.python.org/3/library/timeit.html) in the standard library might be useful here. E.g. something like
```python
    def run_batch():
        unbound.execute_coords_batch(
            coords,
            params,
            boxes,
            lamb,
        )

    mean_time = timeit.timeit("run_batch", number=num_batches)
```
(note this also disables garbage collection in the test loop)

Reply:
The replication is reduced now. Seems cleaner to benchmark this in the same sort of context that we would run it, rather than using `timeit`.